docs/examples/gen_velocities.rst

```diff
@@ -0,0 +1,432 @@
+    * - Velocity component
+      - Symbol
+    * - Earth's velocity as a function of time
+      - :math:`v_\mathrm{E}( t )`
+    * - pulsar's orbital velocity as a function of time
+      - :math:`v_\mathrm{p,orb}( t )`
+    * - pulsar system's barycentric velocity
+        (corresponding to the proper motion)
+      - :math:`v_\mathrm{p,pm}`
+    * - velocity of the interstellar medium at the lens (known in this example)
+      - :math:`v_\mathrm{ISM}`
+
+All these refer to the component of the velocity projected onto the
+one-dimensional lens.
+
+Earth's velocity
+----------------
+
+To obtain Earth's velocity in the lens frame, first generate a location on
+Earth's surface using the :py:class:`~astropy.coordinates.EarthLocation` class
+(in this case the location of the Parkes radio telescope). This class has the
+:py:meth:`~astropy.coordinates.EarthLocation.get_gcrs` method, which returns
+positions (with respect to the centre of the Earth) as a function of time.
+These are transformed into the lens frame using the
+:py:meth:`~astropy.coordinates.BaseCoordinateFrame.transform_to` method.
+Velocities can then be extracted using the
+:py:attr:`~astropy.coordinates.BaseCoordinateFrame.velocity` attribute, and
+finally :py:attr:`~astropy.coordinates.CartesianDifferential.d_z` isolates the
+z-component of the velocity (in the direction of the screen).
+
+.. plot::
+    :include-source:
+    :context:
+
+    earth_loc = EarthLocation('148°15′47″E', '32°59′52″S')
+    
+    v_earth = earth_loc.get_gcrs(t).transform_to(lens_frame).velocity.d_z
+
+Pulsar's orbital velocity
+-------------------------
+
+Compute the pulsar's orbital velocity projected onto the screen
+    
+.. math::
+
+    v_\mathrm{p,orb} = - \frac{ K_\mathrm{p} }{ \sin( i_\mathrm{p} ) }
+                         \left[ \cos( i_\mathrm{p} )
+                                \sin( \mathit{\Delta\Omega}_\mathrm{p} )
+                                \cos( \theta_\mathrm{p} )
+                              - \cos( \mathit{\Delta\Omega}_\mathrm{p} )
+                                \sin( \theta_\mathrm{p} )
+                         \right].
+
+Here, :math:`\theta_\mathrm{p}( t )` is the phase of pulsar orbit as measured
+from its ascending node.
+
+.. plot::
+    :include-source:
+    :context:
+
+    th_p = ((t - t0_p) / p_b).to('') * u.cycle
+
+    v_p_orb = (-k_p / np.sin(i_p)
+               * (np.cos(i_p) * np.sin(delta_omega_p) * np.cos(th_p)
+                              - np.cos(delta_omega_p) * np.sin(th_p)))
+
+Pulsar system barycentre velocity
+---------------------------------
+
+The pulsar system barycentre velocity projected onto the screen is given by
+
+.. math::
+
+    v_\mathrm{p,pm} \simeq d_\mathrm{p}
+                           \left[ \mu_\alpha \sin( \mathit{\Omega}_\mathrm{s} )
+                                + \mu_\delta \cos( \mathit{\Omega}_\mathrm{s} )
+                           \right].
+
+This can be computed manually, but it can also be retrieved from the
```

Comment:
It is somewhat inconsistent to compute `v_p_orb` "manually", but `v_p_pm` using `transform_to(lens_frame)`.

Reply:
I think that's OK. No point calculating it again.